library/src/main/java/com/datatorrent/lib/projection/ProjectionOperator.java

```diff
@@ -0,0 +1,303 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.projection;
+
+import java.lang.reflect.Field;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.lang3.ClassUtils;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+
+import com.datatorrent.common.util.BaseOperator;
+
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * <b>ProjectionOperator</b>
+ * Projection Operator projects defined set of fields from given selectFields/dropFields
+ *
+ * <b>Parameters</b>
+ * - selectFields: comma separated list of fields to be selected from input tuples
+ * - dropFields: comma separated list of fields to be dropped from input tuples
+ * selectFields and dropFields are optional and either of them shall be specified
+ * When both are not specified, all fields shall be projected to downstream operator
+ *
+ * <b>Input Port</b> takes POJOs as an input
+ *
+ * <b>Output Ports</b>
+ * - projected port emits POJOs with projected fields from input POJOs
+ * - remainder port, if connected, emits POJOs with remainder fields from input POJOs
+ * - error port emits input POJOs as is upon error situations
+ * 
+ * <b>Examples</b>
+ * For {a, b, c} type of input tuples
+ *  - when selectFields = "" and dropFields = "", projected port shall emit {a, b, c}
+ *  - when selectFields = "b", projected port shall emit {b} and remainder port shall emit {a, c}
+ *  - when dropFields = "b", projected port shall emit {a, c} and remainder port shall emit {b}
+ * 
+ */
+
+public class ProjectionOperator extends BaseOperator implements Operator.ActivationListener<Context>
+{
+  protected String selectFields;
+  protected String dropFields;
+  protected String condition;
+
+  static class TypeInfo
+  {
+    String name;
+    Class type;
+    PojoUtils.Setter setter;
+    PojoUtils.Getter getter;
+
+    public TypeInfo(String name, Class<?> type)
+    {
+      this.name = name;
+      this.type = type;
+    }
+
+    public String toString()
+    {
+      String s = new String("'name': " + name + " 'type': " + type);
+      return s;
+    }
+  }
+
+  protected transient List<TypeInfo> projectedFields = new ArrayList<>();
```

Comment:
Unless there is a immediate need for a projectedFields & remainderFields to be used by subclass, I'll suggest to make these fields private.